base/pkg/entry.jl

```diff
@@ -148,11 +148,12 @@ function status(io::IO, pkg::AbstractString, ver::VersionNumber, fix::Bool)
 end
 
 function clone(url::AbstractString, pkg::AbstractString)
-    info("Cloning $pkg from $url")
+    normalized_url = Git.normalize_url(url)
```

Comment:
This won't rewrite SSH URLs, only git and https, but you're right, if the user manually provides a URL to clone(), it shouldn't be rewritten.  I'll remove the normalisation from this function.  For reference, the regex (which I've not touched) is:

    r"^(?:git@|git://|https://(?:[\w\.\+\-]+@)?)github.com[:/](([^/].+)/(.+?))(?:\.git)?$"i